packages/ffe-cards-react/src/index.d.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+
+export interface CardBaseProps extends React.HTMLAttributes<HTMLDivElement> {
+    children: React.ReactNode;
+    className?: string;
+    element?: React.ReactNode;
+}
+
+export interface IconCardProps extends CardBaseProps {
+    icon: React.ReactNode;
+    condensed?: boolean;
+    greyCharcoal?: boolean;
+}
+
+export interface ImageCardProps extends CardBaseProps {
+    image: React.ReactNode;
+}
+
+export interface ComponentBaseProps
+    extends React.HtmlHTMLAttributes<HTMLParagraphElement> {
+    className?: string;
+    element?: React.ReactNode;
+    children?: React.ReactNode;
+}
+
+export interface TitleProps extends ComponentBaseProps {
+    overflowEllipsis: boolean;
+}
+
+export interface SubtextProps extends ComponentBaseProps {}
+
+export interface TextProps extends ComponentBaseProps {}
+
+export interface CardNameProps extends ComponentBaseProps {}
+
+export interface TextCardRenderProps {
+    Title: typeof TitleProps;
+    Subtext: typeof SubtextProps;
+    Text: typeof TextProps;
+}
+
+export interface TextCardProps extends CardBaseProps {
+    leftAlign?: boolean;
+    children: React.ReactNode | Function<TextCardRenderProps>;
```

Comment:
Have you tested this Function-definition?

Reply:
I wouldn't make a PR with something I haven't tested ;)

Reply:
That's very good! :)

Reply:
It was a major headache actually 😂 
I have 4 hours of experience with TS so far and the first missing declaration I come across is one of the more complex component APIs we have.

Reply:
Even with a lot more experience, I can imagine :joy: